src/nvim/terminal.c

```diff
@@ -1172,7 +1171,7 @@ static void redraw(bool restore_cursor)
     int off = is_focused(term) ? 0 : (curwin->w_p_rl ? 1 : -1);
     curwin->w_cursor.col = MAX(0, term->cursor.col + win_col_off(curwin) + off);
     curwin->w_cursor.coladd = 0;
-    setcursor();
+    mb_check_adjust_col(curwin);
```

Comment:
@oni-link Seems ok when I tried locally (running `bash`, `top`, `htop`, `nvim`); the tests seem ok too. If the only risk is aesthetic (i.e.: won't corrupt some assumption, leading to a crash), we can try without it, and add it back later if needed.